cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -202,12 +202,14 @@ func (s *CMServer) Run(_ []string) error {
 	}
 
 	if s.AllocateNodeCIDRs {
-		routes, ok := cloud.Routes()
-		if !ok {
+		if cloud == nil {
+			glog.Warningf("allocate-node-cidrs is set, but no cloud provider specified. Will not manage routes.")
+		} else if routes, ok := cloud.Routes(); !ok {
 			glog.Fatal("Cloud provider must support routes if allocate-node-cidrs is set")
```

Comment:
I think it should be okay to set --allocate-node-cidrs without having your cloudprovider support it.
Instead of the glog.Fatal here, what do you think about: 
```glog.Warning("allocate-node-cidrs is set, but cloud provider does not support routes. Will not manage routes.")```

Also, super-nit: The `glog.Warningf` above could just be `glog.Warning`.